drivers/video/isx019.c

```diff
@@ -0,0 +1,3249 @@
+  {
+    100,
+      {
+          1,   1,   1,   1,   1,   1,   1,   1,
+          1,   1,   1,   1,   1,   1,   1,   1,
+          1,   1,   1,   1,   1,   1,   1,   1,
+          1,   1,   1,   1,   1,   1,   1,   1,
+          1,   1,   1,   1,   1,   1,   1,   1,
+          1,   1,   1,   2,   2,   2,   2,   2,
+          2,   2,   2,   2,   2,   2,   3,   3,
+          3,   3,   3,   3,   3,   3,   3,   3,
+      },
+      {
+         64,  64,  64,  64,  64,  64,  64,  64,
+         64,  64,  64,  64,  64,  64,  64,  64,
+         64,  64,  64,  64,  64,  64,  64,  32,
+         64,  64,  64,  64,  64,  64,  32,  32,
+         64,  64,  64,  64,  64,  32,  32,  21,
+         64,  64,  64,  64,  32,  32,  21,  21,
+         64,  64,  64,  32,  32,  21,  21,  21,
+         64,  64,  32,  32,  21,  21,  21,  21,
+      },
+      {
+          1,   1,   1,   1,   1,   1,   1,   1,
+          1,   1,   2,   2,   1,   2,   2,   3,
+          3,   3,   3,   3,   3,   3,   3,   3,
+          3,   3,   3,   3,   3,   3,   3,   3,
+          3,   3,   3,   3,   3,   3,   3,   3,
+          3,   3,   3,   3,   3,   3,   3,   3,
+          3,   3,   3,   3,   3,   3,   3,   3,
+          3,   3,   3,   3,   3,   3,   3,   3,
+      },
+      {
+         64,  64,  64,  64,  32,  21,  21,  21,
+         64,  64,  64,  32,  21,  21,  21,  21,
+         64,  64,  64,  21,  21,  21,  21,  21,
+         64,  32,  21,  21,  21,  21,  21,  21,
+         32,  21,  21,  21,  21,  21,  21,  21,
+         21,  21,  21,  21,  21,  21,  21,  21,
+         21,  21,  21,  21,  21,  21,  21,  21,
+         21,  21,  21,  21,  21,  21,  21,  21,
+      }
+  }
+};
+
+#define NR_JPGSETTING_TBL \
+        (sizeof(g_isx019_jpg_quality) / sizeof(isx019_fpga_jpg_quality_t))
+
+int32_t g_isx019_colorfx[] =
+{
+  IMGSENSOR_COLORFX_NONE,
+  IMGSENSOR_COLORFX_BW,
+  IMGSENSOR_COLORFX_VIVID,
+};
+
+#define NR_COLORFX (sizeof(g_isx019_colorfx) / sizeof(int32_t))
+
+int32_t g_isx019_wbmode[] =
+{
+  IMGSENSOR_WHITE_BALANCE_AUTO,
+  IMGSENSOR_WHITE_BALANCE_INCANDESCENT,
+  IMGSENSOR_WHITE_BALANCE_FLUORESCENT,
+  IMGSENSOR_WHITE_BALANCE_DAYLIGHT,
+  IMGSENSOR_WHITE_BALANCE_CLOUDY,
+  IMGSENSOR_WHITE_BALANCE_SHADE,
+};
+
+#define NR_WBMODE (sizeof(g_isx019_wbmode) / sizeof(int32_t))
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static void i2c_lock(void)
+{
+  nxsem_wait_uninterruptible(&g_isx019_private.i2c_lock);
+}
+
+static void i2c_unlock(void)
+{
+  nxsem_post(&g_isx019_private.i2c_lock);
+}
+
+static void fpga_lock(void)
+{
+  nxsem_wait_uninterruptible(&g_isx019_private.fpga_lock);
+}
+
+static void fpga_unlock(void)
+{
+  nxsem_post(&g_isx019_private.fpga_lock);
+}
+
+static int fpga_i2c_write(uint8_t addr, FAR uint8_t *data, uint8_t size)
+{
+  struct i2c_config_s config;
+  static uint8_t buf[FPGA_I2C_REGSIZE_MAX + FPGA_I2C_REGADDR_LEN];
```

Comment:
Do we really need `static` here? I mean, that I do not see same approach in other places in this file

Reply:
I assume that "resolved" means that `static` is intended to be applied here